tests/ClientTest.php

```diff
@@ -159,6 +159,28 @@ public function testBeforeNotifySkipsError()
         $this->client->notify(Report::fromNamedError($this->config, 'SkipMe', 'Message'));
     }
 
+    public function testBeforeNotifyCanModifyReportFrames()
+    {
+        $this->client = new Client($this->config = new Configuration('example-api-key'), null, $this->guzzle);
+
+        $this->client->setBatchSending(false);
+
+        $this->client->notify($report = Report::fromNamedError($this->config, 'Magic', 'oh no'));
+
+        $this->assertFalse($report->getStacktrace()->getFrames()[0]['inProject']);
+
+        $this->client->registerCallback(function (Report $report) {
+            $frames = &$report->getStacktrace()->getFrames();
+            $frames[0]['inProject'] = true;
```

Comment:
Can we test against something that explicitly won't be there, as `inProject` could conceivably be `true`, making this a false positive (or at least ambiguous to read).  Something like:
`$frames[0]['testFlag'] = true;`
With a corresponding assertion?

Reply:
That's why there are two assertions in this test. One to check that the value is false by default and then adding a callback, notifying, and checking that the value is true.